sky/serve/autoscalers.py

```diff
@@ -132,93 +131,116 @@ def collect_request_information(
                                    current_time - self.rps_window_size)
         self.request_timestamps = self.request_timestamps[index:]
 
-    def evaluate_scaling(
-        self,
-        replica_infos: List['replica_managers.ReplicaInfo'],
-    ) -> AutoscalerDecision:
-
-        num_replicas = len(replica_infos)
-
+    def _get_desired_num_replicas(self) -> int:
+        assert self.target_qps_per_replica is not None
```

Comment:
We should allow this to be None if the user doesn't want autoscaling. IIUC `target_qps_per_replica` is an optional field?